src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/EmailAddressAttribute.cs

```diff
@@ -32,7 +25,10 @@ public override bool IsValid(object value)
             }
 
             var valueAsString = value as string;
-            return valueAsString != null && _regex.Match(valueAsString).Length > 0;
+            return (valueAsString != null
+                && valueAsString.Count(c => c == '@') == 1
+                && valueAsString[0] != '@'
+                && valueAsString[valueAsString.Length-1] != '@');
```

Comment:
You know the scenarios better than I.  If this isn't ever used in a fast path, isn't used repeatedly, etc., then it doesn't matter.  If it is, then it does.